pkg/kubelet/cm/devicemanager/pod_devices.go

```diff
@@ -321,10 +328,69 @@ func (pdev *podDevices) deviceRunContainerOptions(podUID, contName string) *Devi
 			annotationsMap[k] = v
 			opts.Annotations = append(opts.Annotations, kubecontainer.Annotation{Name: k, Value: v})
 		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.DevicePluginCDIDevices) {
+			// Updates for CDI devices.
+			cdiDevices := getCDIDeviceInfo(resp, allCDIDevices)
+			opts.CDIDevices = append(opts.CDIDevices, cdiDevices...)
+		}
+	}
+
+	// Although the CDI devices are expected to be empty when this feature is disabled, we still
+	// guard this with a feature gate to avoid any potential issues.
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.DevicePluginCDIDevices) {
+		cdiAnnotations := getCDIAnnotations(podUID+"-"+contName, sets.List[string](allCDIDevices), annotationsMap)
```

Comment:
You can call `allDvices.UnsortedList()` instead of casting to a `sets.List[string]`. In fact, maybe its better to do this within the body of the function to keep this callsite more compact.

Reply:
I explicitly use `sets.List` to get a sorted list here. One motivation is for odering of the devices in the annotations in the tests.

I can move to a local variable for compactness though.

Reply:
Let's leave the resource ID as is for now (I'm still not a huge fan, but it's not worth bikeshedding over) -- just please add a comment as to what its purpose is.

After that and the change to pull sets.List[string] into the function, I think this PR is good to go.

Reply:
Updated in https://github.com/kubernetes/kubernetes/commit/1d21aa4cc0f7d94463fe867aadf4608317ffbaf0